server/src/concepts/user/routes/index.js

```diff
@@ -1,13 +1,20 @@
-import { createUser } from "../controllers/index.js";
-import { updateUser } from "../controllers/index.js";
-import { deleteUser } from "../controllers/index.js";
-import { loginUser } from "../controllers/index.js";
+import {
+  createUser,
+  updateUser,
+  deleteUser,
+  loginUser,
+} from "../controllers/index.js";
+import {
+  isAuthenticated,
+  isAdmin,
+} from "../../authorization/controllers/auth.js";
 import express from "express";
 
 const router = express.Router();
 
 router.post("/register", createUser);
-router.put("/:id", updateUser);
-router.delete("/:id", deleteUser);
+router.put("/:id", isAuthenticated, isAdmin, updateUser);
```

Comment:
```suggestion
router.put("/:id", isThatUser, updateUser);
```
Here we should use isThatUser to check if user is updating himself, no other user.